ui/lib/apps/ContinuousProfiling/pages/Detail.tsx

```diff
@@ -94,10 +201,28 @@ export default function Page() {
       if (!token) {
         return
       }
-      window.open(
-        `${client.getBasePath()}/continuous_profiling/single_profile/view?token=${token}`,
-        '_blank'
-      )
+
+      if (action === 'view_graph' || action === 'view_text') {
+        const profileURL = `${client.getBasePath()}/continuous_profiling/single_profile/view?token=${token}`
+        window.open(profileURL, '_blank')
+        return
+      }
+
+      if (action === 'view_flamegraph') {
+        // view flamegraph by speedscope
+        const speedscopeTitle = `${rec.target?.component}_${rec.target?.address}_${rec.profile_type}`
+        const profileURL = `${client.getBasePath()}/continuous_profiling/single_profile/view?token=${token}&data_format=protobuf`
+        const speedscopeURL = `${publicPathPrefix}/speedscope#profileURL=${encodeURIComponent(
+          profileURL
+        )}&title=${speedscopeTitle}`
+        window.open(speedscopeURL, '_blank')
+        return
+      }
+
+      if (action === 'download') {
+        window.location.href = `${client.getBasePath()}/continuous_profiling/download?token=${token}`
```

Comment:
@crazycs520 will help implement this API in the ngm side (it doesn't support download single profile before).